pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/TxnLogBufferedWriter.java

```diff
@@ -217,9 +224,9 @@ public void asyncAddData(T data, AddDataCallback callback, Object ctx){
                 // Avoid missing callback, do failed callback when error occur before add data to the array.
                 int recordsCountAfter = dataArray.size();
                 if (recordsCountAfter == recordsCountBeforeAdd){
-                    callback.addFailed(new ManagedLedgerException.ManagedLedgerFencedException(e), ctx);
+                    callback.addFailed(new ManagedLedgerInterceptException(e), ctx);
```

Comment:
I don't understand why you chose the Intercept exception - this is not the error you're witnessing. You tried to asynchronously add data to the buffer and failed. Why not create a new exception if you can't find a similar exception?
@codelipenghui WDYT?

Reply:
Hi @asafm  @codelipenghui 

Since this PR is already too large, I will create a new Exception in the following PR to solve it. Thanks